### PR TITLE
planner/core: migrate test-infra to testify for `planner/core/plan_test.go`

### LINKED ISSUE #28390: migrate test-infra to testify for `planner/core/plan_test.go`

### PR DESCRIPTION
**Pull request by johnhaxx7**
### What problem does this PR solve?

Issue Number: close #28390

Problem Summary:

### Check List

Tests 

- [x] Unit test

Side effects
NA

Documentation
NA

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by johnhaxx7** [comment]
Hi @tisonkun, can you please help take a look when you get a chance?
This is my first PR in TiDB, feel free to let me know if anything needs to be updated, thanks!

File: planner/core/plan_test.go [hunk 1]
```diff
@@ -125,35 +99,38 @@ func (s *testPlanNormalize) TestNormalizedPlan(c *C) {
...
+	defer clean()
+	tk := testkit.NewTestKit(t, store)
 	tk.MustExec("use test")
 	tk.MustExec("drop table if exists t1")
 	tk.MustExec("create table t1 (a int, b int, c int, primary key(a))")
 	tk.MustExec("insert into t1 values(1,1,1), (2,2,2), (3,3,3)")
-
-	tbl, err := s.dom.InfoSchema().TableByName(model.CIStr{O: "test", L: "test"}, model.CIStr{O: "t1", L: "t1"})
-	c.Assert(err, IsNil)
+	dom := domain.GetDomain(tk.Session())
```
Comment (hawkingrei) [inline planner/core/plan_test.go:131]: use ```CreateMockStoreAndDomain``` to get domain.

File: planner/core/plan_test.go [hunk 2]
```diff
@@ -125,35 +99,38 @@ func (s *testPlanNormalize) TestNormalizedPlan(c *C) {
...
+	tbl, err := dom.InfoSchema().TableByName(model.CIStr{O: "test", L: "test"}, model.CIStr{O: "t1", L: "t1"})
+	require.Nil(t, err)
```
Comment (hawkingrei) [inline planner/core/plan_test.go:133]: use ```require.NoError```

Comment (hawkingrei) [reply planner/core/plan_test.go:133]: ditto other

File: planner/core/plan_test.go (hunk 1 above)
Comment (johnhaxx7) [reply planner/core/plan_test.go]: Thanks, resolved!

File: planner/core/plan_test.go (hunk 2 above)
Comment (johnhaxx7) [reply planner/core/plan_test.go]: Thanks, resolved!

**Review by hawkingrei — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Pull request merged**